src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -39,9 +39,14 @@ public String getToken() {
     // Called from other synchronized code. Do not call into other synchronized code or you'll
     // risk deadlock
     public synchronized void setDistinctId(String distinctId) {
-        mUnseenSurveys.clear();
-        mUnseenNotifications.clear();
-        mDistinctId = distinctId;
+        if (mDistinctId.equals(distinctId)) {
+            return; // the user is the same, so no new response is needed
+        } else {
+            mUnseenSurveys.clear();
+            mUnseenNotifications.clear();
+            mVariants = new JSONArray();
```

Comment:
We have to nullify `mVariants` here since it'll conflict with other logic (see https://github.com/mixpanel/mixpanel-android/blob/master/src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java#L179).
Alternatively, we could initialize mVariants in https://github.com/mixpanel/mixpanel-android/blob/master/src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java#L192 and change line 179 to check whether it's empty or not. In here, it'd be better to empty the array instead of creating a new one.